### PR TITLE
IDENTITY-5510: Enabling XACML PIP to register to attribute categories, rather individual attributes

### PR DESCRIPTION
**Pull request by darshanasbg**

### CODE REVIEW

File: components/entitlement/org.wso2.carbon.identity.entitlement/src/main/java/org/wso2/carbon/identity/entitlement/pip/CarbonAttributeFinder.java [hunk 1]
```diff
@@ -136,9 +137,14 @@ public EvaluationResult findAttribute(URI attributeType, URI attributeId, String
 
         List<AttributeValue> attrBag = new ArrayList<AttributeValue>();
         // Get the list of attribute finders who are registered with this particular attribute.
-        List<PIPAttributeFinder> finders = attrFinders.get(attributeId.toString());
 
 
+        List<PIPAttributeFinder> finders = attrFinders.get(category.toString());
```
Comment (prabath) [inline components/entitlement/org.wso2.carbon.identity.entitlement/src/main/java/org/wso2/carbon/identity/entitlement/pip/CarbonAttributeFinder.java]: do a null check on category

File: components/entitlement/org.wso2.carbon.identity.entitlement/src/main/java/org/wso2/carbon/identity/entitlement/pip/CarbonAttributeFinder.java [hunk 2]
```diff
@@ -136,9 +137,14 @@ public EvaluationResult findAttribute(URI attributeType, URI attributeId, String
...
+
+        if (CollectionUtils.isNotEmpty(finders)) {
```
Comment (prabath) [inline components/entitlement/org.wso2.carbon.identity.entitlement/src/main/java/org/wso2/carbon/identity/entitlement/pip/CarbonAttributeFinder.java]: not quite right - it should be, if empty

File: components/entitlement/org.wso2.carbon.identity.entitlement/src/main/java/org/wso2/carbon/identity/entitlement/pip/CarbonAttributeFinder.java [hunk 3]
```diff
@@ -136,9 +138,18 @@ public EvaluationResult findAttribute(URI attributeType, URI attributeId, String
...
+        if (StringUtils.isNotBlank(category.toString())) {
+            finders = attrFinders.get(category.toString());
+        }
+
+        if (CollectionUtils.isEmpty(finders)) {
+            finders = attrFinders.get(attributeId.toString());
+        }
+
         if (finders == null || finders.size() == 0) {
             finders = attrFinders.get(attributeId.toString());
```
Comment (mefarazath) [inline components/entitlement/org.wso2.carbon.identity.entitlement/src/main/java/org/wso2/carbon/identity/entitlement/pip/CarbonAttributeFinder.java]: Why are we trying to get attributeFinders based on attributeId once again? 
Aren't we doing that in line 150?

**Comment by darshanasbg** [comment]
Thanks @prabath and @mefarazath for suggestions. Updated the PR with requested changes.

**Pull request merged**